### DISCUSSION TITLE
Can taiko seleniod run on windows remote client (Not in docker)?

### DISCUSSION THREAD

**Discussion by ShayWeizmanFG** [saikrishna321/taiko-selenoid#262] [General]
Hi I want to run taiko on a remote windows client (Not inside a docker on windows), is it possible with this tool, if not what is the correct way of doing it?

**Comment by saikrishna321** [comment]
You need to start chrome in debug mode in windows machine. openBrowser api you need to specify the windows machine host and port number which the chrome browser is opened.

**Reply by ShayWeizmanFG** [reply]
I need to run tests on the remote machine... why chrome need to be on debug... can't it use chrome driver in some way? Is there an example or documentation about how to run tests on a remote windows machine? I need it for chrome  /  FF / Edge.

**Reply by saikrishna321** [reply]
taiko is chromium based and it does not use any chromedriver. It uses chrome dev tool protocol.

**Reply by ShayWeizmanFG** [reply]
So taiko can run only on chrome dev? what about edge / ff

**Reply by saikrishna321** [reply]
Should work with latest edge as it's chromium based. TAIKO has support for FF. Please see readMe. 

FF SUPPORT IS EXPERIMENTAL

**Reply by ShayWeizmanFG** [reply]
I have opened chrome in dev mode.. and set the host and IP in the node_modules.../index.js to the one I need but I get connection refuse.

**Comment by ShayWeizmanFG** [comment]
How do I pass to the taiko-selenoid the remote host and port? it always try to access 127.0.0.1:4444 although I try to use other host and IP.

My code is like this:
await openBrowser({
            headless: false,
            args: ["--proxy-server=http://" + getenv("TENANT_DOMAIN_NAME") + ":8080"],
            host: '10.190.130.251',
            port: 9000,
            ignoreCertificateErrors: true
        });

**Reply by saikrishna321** [reply]
Taiko-selenoid has api setConfig

**Reply by ShayWeizmanFG** [reply]
setConfig doesn't set host and port:
export interface GlobalConfigurationOptions {
  navigationTimeout?: number;
  observeTime?: number;
  retryInterval?: number;
  retryTimeout?: number;
  noOfElementToMatch?: number;
  observe?: boolean;
  waitForNavigation?: boolean;
  waitForEvents?: BrowserEvent[];
  ignoreSSLErrors?: boolean;
  headful?: boolean;
  criConnectionRetries?: number;
  firefox?: boolean;
  highlightOnAction?: 'true' | 'false';
  local?: boolean;
}

**Reply by saikrishna321** [reply]
Please refer this https://github.com/saikrishna321/taiko-selenoid/blob/master/src/index.spec.js

**Reply by ShayWeizmanFG** [reply]
I still don't understand how to set IP and Host to the selenoid... can you give an example? What need to be import?
